icubam/www/handlers/upload_csv.py

```diff
@@ -2,24 +2,36 @@
 import tornado.web
 import logging
 import time
+from icubam.www.handlers import base
 
-@tornado.web.authenticated
-class UploadHandler(tornado.web.RequestHandler):
+
+class UploadHandler(base.BaseHandler): #tornado.web.RequestHandler):
   """Accept file uploads over POST."""
 
   ROUTE = "/upload_csv"
+  API_COOKIE = 'api'
+
+  def get_current_user(self):
```

Comment:
i think it would be better to add a APIKeyHandler that inherits from BaseHandler and that redefines this exact function and then use it for DB, Map, and Upload. This way u do not need to duplicate code

Reply:
Agreed, let's do this on a separate PR.